cpp/src/arrow/flight/test_integration.cc

```diff
@@ -103,10 +103,142 @@ class AuthBasicProtoScenario : public Scenario {
   }
 };
 
+class TestServerMiddleware : public ServerMiddleware {
```

Comment:
Could you perhaps briefly describe in comments those middleware classes?

Reply:
I added comments to explain the purpose and scope of the test that these classes implement.